data-pipeline/src/trace_exporter.rs

```diff
@@ -1187,70 +1192,71 @@ mod tests {
         exporter.shutdown(None).unwrap();
 
         mock_traces.assert();
-        mock_stats.assert();
+        //mock_stats.assert();
     }
 
-    #[cfg_attr(miri, ignore)]
-    #[test]
-    fn test_shutdown_with_timeout() {
-        let server = MockServer::start();
-
-        let mock_traces = server.mock(|when, then| {
-            when.method(POST)
-                .header("Content-type", "application/msgpack")
-                .path("/v0.7/traces");
-            then.status(200).body("");
-        });
-
-        let _mock_stats = server.mock(|when, then| {
-            when.method(POST)
-                .header("Content-type", "application/msgpack")
-                .path("/v0.6/stats");
-            then.delay(Duration::from_secs(10)).status(200).body("");
-        });
-
-        let mock_info = server.mock(|when, then| {
-            when.method(GET).path("/info");
-            then.status(200)
-                .header("content-type", "application/json")
-                .header("datadog-agent-state", "1")
-                .body(r#"{"version":"1","client_drop_p0s":true}"#);
-        });
-
-        let builder = TraceExporterBuilder::default();
-        let exporter = builder
-            .set_url(&server.url("/"))
-            .set_tracer_version("v0.1")
-            .set_language("nodejs")
-            .set_language_version("1.0")
-            .set_language_interpreter("v8")
-            .set_input_format(TraceExporterInputFormat::V04)
-            .set_output_format(TraceExporterOutputFormat::V07)
-            .enable_stats(Duration::from_secs(10))
-            .build()
-            .unwrap();
-
-        let trace_chunk = vec![pb::Span {
-            duration: 10,
-            ..Default::default()
-        }];
-
-        let data = rmp_serde::to_vec_named(&vec![trace_chunk]).unwrap();
-
-        // Wait for the info fetcher to get the config
-        while mock_info.hits() == 0 {
-            exporter.runtime.block_on(async {
-                sleep(Duration::from_millis(100)).await;
-            })
-        }
-
-        exporter.send(data.as_slice(), 1).unwrap();
-        exporter
-            .shutdown(Some(Duration::from_millis(500)))
-            .unwrap_err(); // The shutdown should timeout
-
-        mock_traces.assert();
-    }
+    // TODO: bring this test back with client side stats
```

Comment:
Not the end of the world since we need to revisit CSS ASAP, but generally we should have a jira ticket assigned for all TODOs. It doesn't guarantee the TODO gets addressed, but it increases the chances.

Reply:
Yeah good call out, just added one